placeSpecificContent/hq/courtyard/controlPanel/controlPanel_server.js

```diff
@@ -0,0 +1,406 @@
+                var courtyardHeight = courtyardProperties.dimensions.y;
+                
+                // if the top of the table is below the top of the courtyard, reposition it to be just the top of 
+                // the courtyard. Using the top of the courtyard gives a little leeway to ensure the table ends up below it
+                if (currentPosition.y + tableHeight * HALF <= courtyardProperties.position.y + courtyardHeight * HALF) {
+                    currentPosition = {
+                        x: currentPosition.x,
+                        y: courtyardProperties.position.y - tableHeight * HALF,
+                        z: currentPosition.z
+                    };
+                    Entities.editEntity(roundtable, { position: currentPosition });
+                }
+            }
+            
+            var distanceToTarget = currentPosition.y.toFixed(FIXED_DECIMAL_PLACES) - 
+                targetPosition.y.toFixed(FIXED_DECIMAL_PLACES);
+            if (distanceToTarget.toFixed(FIXED_DECIMAL_PLACES) === 0 || 
+                Math.abs(distanceToTarget.toFixed(FIXED_DECIMAL_PLACES)) < TARGET_DEVIATION_ALLOWANCE) {
+                Entities.editEntity(entityID, {
+                    position: targetPosition
+                });
+                if (entityID === roundtable) {
+                    roundtableInPlace = true;
+                } else {
+                    courtyardInPlace = true;
+                }
+                return;
+            }
+            
+            // If the entity has somehow been moved far away, slam it into lowered position to avoid hitting anyone
+            if (Math.abs(distanceToTarget) > MAX_DISTANCE_TO_TRANSITION_ENTITIES_M && 
+                Math.abs(distanceToTarget) < -MAX_DISTANCE_TO_TRANSITION_ENTITIES_M) {
+                currentPosition = entityID === roundtable ? roundtableLoweredPosition : courtyardLoweredPosition;
+                Entities.editEntity(entityID, {
+                    position: currentPosition
+                });
+            // If entity is very close to target position, slam into exact position
+            } else if (Math.abs(distanceToTarget) < MIN_DISTANCE_TO_TRANSITION_ENTITIES_M && 
+                Math.abs(distanceToTarget) > -MIN_DISTANCE_TO_TRANSITION_ENTITIES_M) {
+                currentPosition = targetPosition;
+                Entities.editEntity(entityID, {
+                    position: targetPosition
+                });
+                if (entityID === roundtable) {
+                    roundtableInPlace = true;
+                } else {
+                    courtyardInPlace = true;
+                }
+            }
+            // If entity is not aligned along x and z axes, slam into correct position on those axes
+            if (currentPosition.x.toFixed(FIXED_DECIMAL_PLACES) !== targetPosition.x.toFixed(FIXED_DECIMAL_PLACES) || 
```

Comment:
It's dangerous to compare mutable floating point values to other floating point values exactly. Please use some small epsilon for comparison instead of `===`.

Reply:
https://github.com/highfidelity/hifi-content/pull/462#discussion_r325276438